server/src/main/java/org/elasticsearch/cluster/metadata/SearchEngine.java

```diff
@@ -92,8 +100,9 @@ public SearchEngine(StreamInput in) throws IOException {
             in.readList(Index::new),
             in.readBoolean(),
             in.readBoolean(),
-            in.readOptionalString(),
-            in.readOptionalString()
+            in.readString(),
```

Comment:
I assume this change was made because we can't have too many optional Strings? If one or more of relevance settings, curations or analytics is omitted from the request will this break?

Reply:
In fact readString is able to read null without any problem. So it is OK to use it and it will not break if not set.